### PR TITLE
Ft/payment wallet option/151338403

### PR DESCRIPTION
**Pull request by THEozmic**
#### What does this PR do?
Add wallet checkout and dashboard feature
#### Description of Task to be completed?
A user should be able to checkout using wallet, transfer funds to other users' wallets and fund their wallet
#### How should this be manually tested?
Login, click on "wallet" from the dropdown, provide amount and click on "submit" to fund wallet.
Provide email and amount, click on "transfer" to transfer.
To pay with wallet, click on wallet payment option, click on "pay now"
#### What are the relevant pivotal tracker stories?
#151338403
#### Screenshots (if appropriate)
image
image

### CODE REVIEW

File: imports/plugins/core/layout/client/templates/layout/wallets/wallet.html [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+              <input id="transferAmount" class="form-control" type="number" name="amount" placeholder="Amount" required>
+              <input id="recipient" class="form-control" type="text" name="email" placeholder="Beneficiary Email" required>
+              <button type="submit" style="margin-top:10px" class="form-control btn btn-success">Transfer</button>
+            </div>
+          </form>
+        </div>
+      </div>
+    </div>
+  </div>
+</template>
```
Comment (mradeybee) [inline imports/plugins/core/layout/client/templates/layout/wallets/wallet.html]: Add an empty line at the end of the file.

File: imports/plugins/core/layout/client/templates/layout/wallets/wallet.js [hunk 2]
```diff
@@ -0,0 +1,300 @@
...
+  document.getElementById("last").disabled = currentPage === numberOfPages
+    ? true
+    : false;
+}
+
+function loadList() {
+  list = Template.instance().state.get("transactionsList");
+  let begin = (currentPage - 1) * numberPerPage;
+  let end = begin + numberPerPage;
+  if (list[0] !== undefined) {
```
Comment (mradeybee) [inline imports/plugins/core/layout/client/templates/layout/wallets/wallet.js]: you might want to have `undefined` as a string. I am not totally sure, but I think its safer.

File: imports/plugins/core/layout/client/templates/layout/wallets/wallet.js [hunk 3]
```diff
@@ -0,0 +1,300 @@
...
+}
+
+function check() {
+  document.getElementById("next").disabled = currentPage === numberOfPages
+    ? true
+    : false;
+  document.getElementById("previous").disabled = currentPage === 1
+    ? true
+    : false;
+  document.getElementById("first").disabled = currentPage === 1 ? true : false;
```
Comment (mradeybee) [inline imports/plugins/core/layout/client/templates/layout/wallets/wallet.js]: You dont need to write this like this. `currentPage === 1` already returns true or false, so, just do ` document.getElementById("first").disabled = currentPage === 1` and you will be good. Apply this to all other places.

File: imports/plugins/core/layout/client/templates/layout/wallets/wallet.js [hunk 4]
```diff
@@ -0,0 +1,300 @@
...
+
+
+const getPaystackSettings = () => {
+  const paystack = Packages.findOne({
+    name: "paystack",
+    shopId: Reaction.getShopId()
+  });
+
+  return {
+    public: "pk_test_3e07adc39d17fdb204c221ed2b2319ed6267d4df",
```
Comment (mradeybee) [inline imports/plugins/core/layout/client/templates/layout/wallets/wallet.js]: Put keys on env variables.

**Review by mradeybee — CHANGES_REQUESTED** [review]
Had few comments and please look at what is making hound complain about ES6 stuffs.

File: imports/plugins/core/layout/client/templates/layout/wallets/wallet.js [hunk 5]
```diff
@@ -0,0 +1,296 @@
...
+          method: "Paystack",
+          transactionId: paystackResponse.reference,
+          currency: paystackResponse.currency,
+          amount: parseInt(paystackResponse.amount, 10),
+          status: paystackResponse.status,
+          mode: "authorize",
+          createdAt: new Date()
+        };
+        if (type === "deposit") {
+          console.log(paystackResponse.amount, "bleh bleh");
```
Comment (mradeybee) [inline imports/plugins/core/layout/client/templates/layout/wallets/wallet.js]: Remove log

**Review by mradeybee — CHANGES_REQUESTED** [review]
This PR looks very identical to https://github.com/andela/elrond-rc/pull/19 what is going on?

**Comment by THEozmic** [comment]
I accidentally checkout out from this branch

File: imports/plugins/core/layout/client/templates/layout/wallets/wallet.js (hunk 5 above)
Comment (THEozmic) [reply imports/plugins/core/layout/client/templates/layout/wallets/wallet.js]: Done

**Review by mradeybee — APPROVED** [review]

**Pull request merged**